dockertest/subtest.py

```diff
@@ -429,13 +458,61 @@ def try_all_stages(self, name, subsubtest):
                               self.exception_info["error_source"],
                               detail)
             exc_info = self.exception_info["exc_info"]
+            # Ignore known problems in specific docker builds
+            if self.is_known_failure(name):
+                self.logwarning("Treating %s subsubtest failure as PASS", name)
+                self.final_subsubtests.add(name)
             # Treat N/A as passed, even though this may hide obscure failures
             if isinstance(detail, TestNAError):
```

Comment:
Not sure this is a problem or not, just looking at the condition:  If a sub-subtest throws a ``TestNAError`` and that test also happens to satisfy ``is_known_failure()``.  The code will:

*  Log the traceback from the test
* Log it's a known failure, "treating it as a pass"
* Log, "Treating TestNAError as a PASS".

Seeing that, one question could be- Is this test Not applicable (to this environment / setup) or is it expected to fail?

Wondering if maybe the "TestNAError" should take priority...like:

```python
    if isinstance(detail, TestNAError):
        ...
    elif if self.is_known_failure(name):
        ...
    else:
        raise exc_info[0], exc_info[1], exc_info[2]
```

???
```